CefSharp.Core/Internals/ClientAdapter.cpp

```diff
@@ -290,6 +307,28 @@ namespace CefSharp
             }			
         }
 
+        void ClientAdapter::OnResourceRedirect(CefRefPtr<CefBrowser> browser, CefRefPtr<CefFrame> frame, const CefString& oldUrl, CefString& newUrl)
+        {
+            IRequestHandler^ handler = _browserControl->RequestHandler;
+            if (handler != nullptr)
+            {
+                auto managedNewUrl = StringUtils::ToClr(newUrl);
+                handler->OnResourceRedirect(_browserControl, frame->IsMain(), StringUtils::ToClr(oldUrl), managedNewUrl);
+
+                //TODO: Determine if there is a better way to pass the reference in rather than reassigning the string
+                newUrl = StringUtils::ToNative(managedNewUrl);
```

Comment:
The same way you'd pass any other non-value type into a .Net method. Isn't the C++/CLI syntax for that the % thingy or something? I'd have to look it up in the C++/CLI syntax.

Other than that, no, `StringBuilder` is as good as it gets. :frowning: 

Bill